src/metabase/sync.clj

```diff
@@ -34,6 +34,22 @@
      [:name       :string]
      [:steps      [:maybe [:sequential sync-util/StepNameWithMetadata]]]]]])
 
+(def ^:private phase->fn
+  {:metadata     sync-metadata/sync-db-metadata!
+   :analyze      analyze/analyze-db!
+   :field-values field-values/update-field-values!})
+
+(defn- scan-phases [scan]
+  (if (not= :full scan)
+    [:metadata]
+    [:metadata :analyze :field-values]))
+
+(defn- do-phase! [phase database]
+  (let [f      (phase->fn phase)
+        result (f database)]
+    (when-not (instance? Throwable result)
```

Comment:
`result` is only allowed to be a `Throwable` if `*log-exceptions-and-continue?*` is `true`. If it's a `Throwable` and `*log-exceptions-and-continue?*=true` that would be a bug and we should definitely not swallow it. Maybe make that more explicit here, like this:

```suggestion
    (when-not (and *log-exceptions-and-continue?* (instance? Throwable result))
```

Reply:
Good thinking

Reply:
Rather than throwing an unrelated type error though, I guess the right thing to do will be to rethrow.